### ISSUE TITLE
Research DHS, MICS APIs for Gisida Ingestion 

### ISSUE THREAD

**Issue by cKellyDesign** [onaio/gisida#188]
As a Gisida client I should be able to connect layers data sources to DHIS and MICS data.

This ticket tracks research outcomes, insights, and conversations.

**Comment by cKellyDesign** [comment]
## DHIS API
* The API itself is public (I think) so Gisida could call it directly by specifying an API URL as the layer data source
* The API has good [documentation](http://api.dhsprogram.com/#/index.html) on all of it's [endpoints](http://api.dhsprogram.com/#/introapi.cfm) and can be accessed as REST or Query based formatted URLs (`http://api.dhsprogram.com/rest/dhs/[ENDPOINT]/[PARAM_VALUE]` is the same as `http://api.dhsprogram.com/rest/dhs/[ENDPOINT]?[PARAM_KEY]=[PARAM_VALUE]`)
* The API is flexible and can be utilized in a way to dynamically load DHIS data (`http://api.dhsprogram.com/rest/dhs/[ENDPOINT]/[PARAM_1],[PARAM_2],[PARAM_N]`)
* API response i paginated (100 records per page) but if we [pass in our API key](http://api.dhsprogram.com/#/api-advancedqueries.cfm) the we can get responses with more reports per page, otherwise we can deal with pagination
* API can return JSON (by default) or CSV formatted data, JSON will probably be easier

### Endpoints
Countries - http://api.dhsprogram.com/rest/dhs/countries
Indicators - http://api.dhsprogram.com/rest/dhs/indicators
Surveys - http://api.dhsprogram.com/rest/dhs/surveys
Data - http://api.dhsprogram.com/rest/dhs/data

### Next Steps
1) Immediate - Attempt a country level layer in Service Mapping (will require a country level geojson file or shapefile/vector tileset to join with for fill layers)
2) Short Term - If 1 doesn't work, use the API endpoints to build out static CSV/JSON files for Gisida layers to be loaded in normally
3) Long Term - Build a DHIS connector module in Gisida to ingest API responses for layers

**Comment by cKellyDesign** [comment]
## MICS
I cannot seem to find an API for this data nor can I find any data output or downloads other than for statistical analysis.

Survey List - http://mics.unicef.org/surveys
MICS Data Tools - http://mics.unicef.org/tools?round=mics5#data-processing

@csc5k @engwiri - Do we need to produce CSV or JSON outputs from the datasets we've been given access to?? I hope not because that sounds like a lot of manual work and requires knowing [SAS/STAT](https://www.sas.com/en_us/software/stat.html)

This doesn't seem like it would be live data.

**Comment by csc5k** [comment]
@cKellyDesign DHS - I found a "RegionId" from the first API listed here: http://api.dhsprogram.com/#/introapi.cfm

I also see a "RegionId" field from the query-based API output: http://api.dhsprogram.com/rest/dhs/data?countryIds=KE&f=xml

I think that might give us the subnational breakdowns

Regarding MICS - I'm not sure what you mean about producing CSV or JSON outputs. I don't think we need to have the Gisida site produce that FOR the client, but we do need to translate SAS/STAT formats into CSV so that Gisida can use it. Is that what you mean?

**Comment by engwiri** [comment]
@cKellyDesign i think what we need immediately is to understand feasibility and possible level of effort. It is fantastic that DHS has an API thats a great first step. Do you know if you can access shape files? or would that have to be separate either way i don't think thats a big deal. We do need to have a sense of how much it would cost to build the long term solution for country and sub national levels

**Comment by cKellyDesign** [comment]
@csc5k - Yeah I saw the `RegionId` but I think that is actually multiple countries per region (at least that was my understanding without looking into it further). As for MICS, correct Gisida needs the CSV to work.

@engwiri - I didn't try pulling the geographic data along with the indicator data since I figured we could just get the shapefiles from HDX since we will need to combine them anyways. 

As for LOE of a DHIS connector module in Gisida I estimate about 5-8 days of engineering work.
